### PR TITLE
feat(vdp): add server plugin to enable SSE streaming and enable endpoints and configuration

### PR DESCRIPTION
**Pull request by tillknuesting**
Because:
- KrakenD does not naively support long-lived connections, such as SSE or any other form of streaming.

This commit:
- implement a server plugin for KrakenD that intercepts HTTP requests for the SSE endpoint and connects to the backed-end to transmit data
- adds endpoint and configuration to support streaming

### CODE REVIEW

**Comment by tillknuesting** [comment]
for reference here is the standalone PoC for the plugin: https://github.com/tillknuesting/krakend-streaming

File: config/share/settings-env/input_headers.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "Instill-User-Id",
     "Instill-Visitor-Uid",
     "Instill-Return-Traces",
-    "Instill-Share-Code"
+    "Instill-Share-Code",
+    "X-Use-SSE"
```
Comment (donch1989) [inline config/share/settings-env/input_headers.json:30]: We currently use the `Instill-` prefix for our custom headers and should maintain this convention.

Comment (donch1989) [reply config/share/settings-env/input_headers.json:30]: Hi @tillknuesting ,
I noticed that you changed the header to `Instill-Use-SSE` in the pipeline-backend, so we need to update it here accordingly.

File: plugins/sse-streaming/main.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		// TODO: Performance optimize matchStrings; critical, every request to the API gateway uses this.
+		matchPaths, id := matchStrings(endpoint, req.URL.Path)
+
+		if !matchPaths {
+			h.ServeHTTP(w, req)
+			return
+		}
+
+		// Construct serverURL using the extracted ID
+		serverURL := fmt.Sprintf("%s%s", backendHost, strings.Replace(backendURLPattern, "{id}", id, 1))
```
Comment (donch1989) [inline plugins/sse-streaming/main.go:92]:
```suggestion
		serverURL := fmt.Sprintf("http://%s%s", backendHost, strings.Replace(backendURLPattern, "{id}", id, 1))
```
I just tried, I guess we need to add `http` here.

**Review by donch1989 — CHANGES_REQUESTED** [review]

**Review by donch1989 — APPROVED** [review]
LGTM

**Pull request merged**